crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -1273,8 +1279,17 @@ impl<InstalledPackages: InstalledPackagesProvider> ResolverState<InstalledPackag
             index,
             env,
             self.tags.as_ref(),
+            exclude_newer,
         ) else {
             // Short circuit: we couldn't find _any_ versions for a package.
+            // Log exclude_newer message once per package across all forks.
+            if let Some(exclude_newer_ts) = exclude_newer {
+                if self.logged_exclude_newer.insert(name.clone()) {
+                    debug!(
+                        "Excluding candidates for {name} published after {exclude_newer_ts}"
+                    );
```

Comment:
Is this being logged if exclude newer is set and we visit this package? That seems misleading? Shouldn't we only log it if we actually filtered a distribution due to exclude newer?

Why should we have this in a debug log rather than a hint on failed resolution?